### DISCUSSION TITLE
Is there any provision to dynamically schedule the workflow via cron

### DISCUSSION THREAD

**Discussion by SachinMehndiratta** [elsa-workflows/elsa-core#937] [Q&A]
Hello,

Can anyone suggest how to dynamically schedule a workflow preferrably using quartz in Elsa2.

What I am looking for is the ability via UI to schedule the worflows. App users can schedule the workflow as per their choice.

Thanks in advance

**Comment by sfmskywalker** [comment]
Easy:

1. Create your (ASP).NET Core project.
2. Add the necessary services (`.AddQuartzTemporalActivities` - make sure to reference the `Elsa.Activities.Temporal.Quartz` package).
3. Create your workflow.
4 Add the `Cron` activity to your workflow and configure it with the desired cron expression.
5. Connect any subsequent activities to perform work.

For an example of setting up an ASP.NET Core project that acts as both the server and the dashboard, checkout [this sample project](https://github.com/elsa-workflows/elsa-core/tree/feature/elsa-2.0/src/samples/dashboard/aspnetcore/ElsaDashboard.Samples.AspNetCore.Monolith).

Ideally, you host your workflows server separate from your dashboard application. But, that's up to you.

**Reply by SachinMehndiratta** [reply]
Thanks @sfmskywalker  for this explanation, however I am a bit confused regarding scheduling a worfklow at runtime via code

I am not looking for any dashboard interaction as of now. Goal is to allow the users to schedule/reschedule/unsubscribe the workflow via Angular app at their convenient times. In my existing app Users  provide Cron expression from angular UI to my backend API and now I want backend API to interact with ELSA for scheduling the workflow

**Your step 3 says - Create a workflow**  I doubt here we could have 2 possibilities
 - Create and register a static workflow definition  i.e. compile time and at the runtime instantiate(schedule/reschedule) the same for each user according to their preferences. I feel this should be the way but not sure how to do this. For me Cron is not working, not even when workflow started via ConfigureServices. 
 
Write method is called only once.
 
       public void Build(IWorkflowBuilder builder)
        {
            builder
                .Cron("* 0 0 ? * * *").Then(Write);
        }

        private async ValueTask Write(ActivityExecutionContext context)
        {
             //Sometask here
        }

I tried the below code for instantiating on demand, again this get executed only once
       
         var startupRunner = services.GetRequiredService<IStartupRunner>();
         await startupRunner.StartupAsync();

         // Get a workflow runner.
         var workflowRunner = services.GetRequiredService<IBuildsAndStartsWorkflow>();

         // Run the workflow.
         await workflowRunner.BuildAndStartWorkflowAsync<CronTaskWorkflow>();
                
         

 - Create a new dynamic workflow definition at run time via code every time user schedule that piece of functionality.

I am looking into ELSA for the first time and not sure if I am making sense in my above points.

If there is any sample demonstrating the concept for scheduling/rescheduling a workflow by n number of users at their own choice of time and requirement.

Thanks again for your answer on this.

**Reply by PKEnergy** [reply]
Have you found any solution for dynamic workflow planning/scheduling?
